parlai/core/teachers.py

```diff
@@ -653,6 +655,13 @@ def __init__(self, opt, data_loader=None, cands=None, shared=None, **kwargs):
             self.data = []
             self._load(data_loader, opt['datafile'])
             self.cands = None if cands is None else set(sys.intern(c) for c in cands)
+
+        self.rank = get_rank()
+        self.num_workers = num_workers()
+        self.is_distributed_and_is_eval = is_distributed() and any(
```

Comment:
Leave a comment saying "We only shard the data set at evaluation time, as training is done using sampling-with-replacement."